Optional face detect

SumwunDucainest opened the pull request:
As discussed on my previous MR, this makes the face detection feature optional at runtime by giving the user a toggle to turn it on or off.

Comments welcome, of course!

SumwunDucainest committed:
Make face detection optional

The current implementation of face detection is WIP, which can lead to a distracting rectangle in the ViewFinder.
This commit makes that feature optional at compile time using #defines and disables it by default.
It can be enabled by adding FACE_DETECTION to DEFINES in harbour-shutter.pro.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
Revert "Make face detection optional"

This reverts commit e8919200af75ab48c5f26738790342643c048554.

SumwunDucainest committed:
Make face detection optional: Add settings to camera proxy

This commit adds a handle to a settings object on Camera Proxy, in order for it
to be able to look up the value of the `faceDetection` setting, which will be used
to enable/disable the face detection feature at runtim.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
Make face detection optional: Create the settings

This commit adds the Settings object that will be shared between the QML
frontend and the C++ backend of the app, and give its handle to the CameraProxy.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
Make face detection optional: Add a button

This commit adds a face detection toggle in the general panel of the settings
for the user to be able to turn face detection on and off at runtime.

Note that because the settings so far were on the QML side only, the setting for
this button currently lives in a different object.
Ultimately, we might want to merge both setting scopes into one singl shared one.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
Make face detection optional: comments

This commit simply adds some comments to clarify the role of m_rects.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

piggz commented:
Hi.  The only change i would make would be to re-do how the settings are handled.  There is already a mechanism for the settings, and you have added another.  I think you just need to add a global property to the Settings{{} object in the application qml file, and use the methods there, no need to add a new appSettings context.

SumwunDucainest committed:
QML: Remove unused function "getCameraValue"

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
QML: Remove unused function "setCameraValue"

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
QML: Remove "getGlobalValue"

The getGlobalValue was only used as a shortcut to a get("global",...
,...) call within harbour-shutter.qml.
If we only use the native C++ Settings object, this method is no longer
needed.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
QML: Move properties out of settings

This commit moves all properties (such as cameraId, captureMode, ...)
out of the `settings` scope and up into the `app` scope in anticipation
of removing the `settings` object.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
QML: Move sizeToStr and strToSize up

This commit moves both helper functions sizeToStr and strToSize up one
scope level from `settings` to `app` so that they are available
everywhere.
It also replaces to re-implementations in lower scopes, in the
AboutVideo and AboutImage pages.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
QML: move "setGlobalValue" from `settings` to `app`

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
QML: Move camera-specific values getter and setter

This commit moves the setCameraModeValue and getCameraModeValue
functions up to `app` from `settings`.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
QML: Move the resolution function

This commit moves the function resolving the appropriate resolution for
a given mode from `settings` to `app`.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
QML: Use the app.calculateEnabledCameras function

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>

SumwunDucainest committed:
Settings: Merge settings and appSettings

This commit gets rid of the `appSettings` object shared between C++ and
QML, and replaces the QML `settings` object by renaming `appSettings`.
This meanns, that the C++ side will now have a handle on the settings
object used by QML.

Signed-off-by: Sumwun Ducainest <sumwun.ducainest@outlook.com>
